### DISCUSSION TITLE
New tip: Using Multipass to create multiple Ubuntu VM for typeC CML mining on your Mac or Windows machine

### DISCUSSION THREAD

**Discussion by kevingzhang** [tearust/teaproject#92] [General]
Since epoch5, type C CML can start AA (Availability Attestation)mining by hosted in a local (non public IP ) machine.

Because the CML require a Ubuntu Linux to run. Most of users do not have a Linux machine at home. Renting a cloud linux server doesn't make sense since no public IP is required.

So the best solution would be running Ubuntu VMs. There are many choices, after a comparison, the easiest one would be using Multipass (https://multipass.run/)

Using Multipass you can use command line to launch and manage Ubuntu VM on top of your existing Mac or Windows as if you are using a Docker.

https://multipass.run/ is the web site.

Because type C CML doesn't need public IP. You can technically run multiple Ubuntu VM on the same Mac/Windows computer and plant multiple type C CML to it. You can earn more revenue without increasing machine costs.

Note, every CML will need 2G RAM. Make sure your machine has enough remaining memory.

@geopor , can you write this formally into the wiki

**Comment by geopor** [comment]
Ok added and also included a link to it from the original mining hardware setup guide, https://github.com/tearust/teaproject/wiki/Mining-With-Your-Own-Hardware---Virtual-Machines
